sources/cfsignal/src/main.rs

```diff
@@ -0,0 +1,97 @@
+#![deny(rust_2018_idioms, unused_imports)]
+
+/*!
+## Introduction
+
+Cfsignal is similar to [cfn-signal] (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-signal.html).
+
+When creating an Auto Scaling Group, CloudFormation can be configured to wait for the expected number of signals from instances before considering the ASG successfully created. See [CreationPolicy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-creationpolicy.html) and [UpdatePolicy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-updatepolicy.html) for more details.
+
+Cfsignal uses `systemctl is-system-running` to determine whether the boot has succeeded or failed, and sends the corresponding signal to the CloudFormation stack.
+
+## configuration
+
+Cfsignal is configured using the following settings:
+* `settings.cloudformation.should-signal`: Check status and send signal. Defaults to `false`.
+* `settings.cloudformation.stack-name`: Name of the stack to signal.
+* `settings.cloudformation.logical-resource-id`: The logical ID of the AutoScalingGroup resource that you want to signal.
```

Comment:
It would be nice if these settings also get documented in the main Bottlerocket README's settings section here betwen "Amazon ECS Settings"  and "OCI Hooks settings":
https://github.com/bottlerocket-os/bottlerocket/blob/9749af9973e4494768c2fac563172c753b6bc011/README.md?plain=1#L420-L422

Something like:
```
#### CloudFormation signal helper settings

For AWS variants, these settings allows you to set up CloudFormation signalling to indicate whether Bottlerocket hosts running in EC2 have been successfully created or updated:
* `settings.cloudformation.should-signal`: Check status and send signal. Defaults to `false`.
* `settings.cloudformation.stack-name`: Name of the stack to signal.
* `settings.cloudformation.logical-resource-id`: The logical ID of the AutoScalingGroup resource that you want to signal.
```

Then the README here in `cfsignal` can reference the settings via a link and instead describe the toml file settings:
```suggestion
Configuration is read from a TOML file, which is generated from Bottlerocket settings:
`should_send`: Whether to check system status and send signal.
`stack_name`: Name of the CFN stack to signal.
`logical_resource_id`: The logical ID of the AutoScalingGroup resource that you want to signal.
```

Feel free to play around with the wording a bit.